scm/driver/stash/repo.go

```diff
@@ -137,10 +137,10 @@ func (s *repositoryService) FindPerms(ctx context.Context, repo string) (*scm.Pe
 		}, nil, nil
 	}
 	// HACK: test if the user has write access to the repository.
-	_, name := scm.Split(repo)
+	namespace, _ := scm.Split(repo)
 	repos, _, _ := s.listWrite(ctx, repo)
 	for _, repo := range repos {
-		if repo.Name == name {
+		if repo.Namespace == namespace {
```

Comment:
this code fix looks cool, one thing is that this was a bug where by the namespace and key were different, can we make sure that this is checked in the unit test https://github.com/drone/go-scm/blob/master/scm/driver/stash/repo_test.go#L142-L186

Reply:
Done.